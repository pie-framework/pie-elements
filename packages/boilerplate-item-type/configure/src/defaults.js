// Should be exactly the same as controller/defaults.js
export default {
  model: {
    prompt: 'Question Prompt goes here',
    promptEnabled: true
  },
  configuration: {
    prompt: {
      label: 'Prompt',
      settings: true,
    }
  },
};
