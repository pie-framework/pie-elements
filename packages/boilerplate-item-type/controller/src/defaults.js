// Should be exactly the same as configure/defaults.js
export default {
  model: {
    prompt: '',
    promptEnabled: true,
  },
  configuration: {
    prompt: {
      label: 'Prompt',
      settings: true,
    },
  },
};
