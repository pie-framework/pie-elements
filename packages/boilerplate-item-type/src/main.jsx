import React from 'react';
import PropTypes from 'prop-types';
import { PreviewPrompt } from '@pie-element/pie-utils/render-ui';
import UTILS from '@pie-element/pie-utils';

const {test, callTest} = UTILS;

callTest('player');
export default class Main extends React.Component {
  static propTypes = {
    model: PropTypes.object,
    session: PropTypes.object,
    onSessionChange: PropTypes.func.isRequired,
  };

  static defaultProps = {};

  render() {
    const { model } = this.props;

    return (
      <div>
        {model.prompt && (
          <div>
            <p>{test}</p>
            <p>Boilerplate Item</p>
            <PreviewPrompt prompt={model.prompt}/>
          </div>
        )}
      </div>
    );
  }
}
