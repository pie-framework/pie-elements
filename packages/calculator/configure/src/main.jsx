import React from 'react';
import PropTypes from 'prop-types';
import { layout, TwoChoice } from '@pie-lib/config-ui';
import { CalculatorLayout } from '@pie-element/calculator';
import Typography from '@material-ui/core/Typography';

const MainTypes = {
  model: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
};

export class Main extends React.Component {
  static propTypes = MainTypes;

  onModeChange = (mode) => {
    const { model, onChange } = this.props;
    const update = { ...model, mode };

    onChange(update);
  };

  render() {
    const { model } = this.props;

    return (
      <layout.ConfigLayout hideSettings={true} settings={null}>
        <TwoChoice
          header="Choose a Calculator"
          value={model.mode}
          onChange={this.onModeChange}
          one={{ label: 'Basic', value: 'basic' }}
          two={{ label: 'Scientific', value: 'scientific' }}
        />
        <br />

        <CalculatorLayout mode={model.mode} onClose={() => ({})} />
        <br />

        <Typography style={{ paddingTop: '4px' }}>
          Please note that the calculators are tools for students and do not record answers.
        </Typography>
      </layout.ConfigLayout>
    );
  }
}

export default class Stateful extends React.Component {
  static propTypes = MainTypes;

  constructor(props) {
    super(props);
    this.state = { model: props.model };
  }

  UNSAFE_componentWillReceiveProps(props) {
    this.setState({ model: props.model });
  }

  onChange = (model) => {
    this.setState({ model }, () => {
      this.props.onChange(this.state.model);
    });
  };

  render() {
    return <Main model={this.state.model} onChange={this.onChange} />;
  }
}
