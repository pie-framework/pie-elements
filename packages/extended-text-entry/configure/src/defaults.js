export default {
  prompt: 'This is the question prompt',
};
