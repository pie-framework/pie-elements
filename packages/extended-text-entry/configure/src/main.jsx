import React from 'react';
import {
  NumberTextField,
  FeedbackSelector,
  InputContainer,
  settings,
  layout
} from '@pie-lib/config-ui';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import EditableHtml from '@pie-lib/editable-html';

const { Panel, toggle } = settings;

const defaultFeedback = {
  type: 'default',
  default: 'Your answer has been submitted'
};

export class Main extends React.Component {
  static propTypes = {
    onChange: PropTypes.func.isRequired,
    model: PropTypes.object.isRequired,
    classes: PropTypes.object.isRequired
  };

  constructor(props) {
    super(props);
    this.state = {
      setDimensions: true
    };
    this.changeWidth = this.change('width');
    this.changeHeight = this.change('height');
  }

  change = key => (event, v) => {
    const { onChange } = this.props;
    const model = this.applyUpdate({ [key]: v });
    onChange(model);
  };

  onPromptChange = markup => {
    const { onChange } = this.props;
    const model = this.applyUpdate({ prompt: markup });
    onChange(model);
  };

  applyUpdate(update) {
    const { model } = this.props;
    const out = Object.assign({}, model, update);
    return out;
  }

  changeFeedback = feedback => {
    const { model, onChange } = this.props;
    const update = { ...model, feedback };
    onChange(update);
  };

  render() {
    const { model, classes, onChange } = this.props;
    const {
      configure: {
        equationEditor,
        multipleParts,
        teacherInstructions,
        studentInstructions,
        mathInput,
        width,
        height
      },
    } = model;

    return (
      <layout.ConfigLayout
        settings={
          <Panel
            model={model}
            onChangeModel={model => onChange(model)}
            groups={{
              'Item Type': {
                mathInput: mathInput.settings && toggle(mathInput.label),
                'configure.equationEditor.enabled': equationEditor.settings &&
                toggle(equationEditor.label),
                'configure.multipleParts.enabled': multipleParts.settings &&
                toggle(multipleParts.label),
              },
              'Properties': {
                'configure.teacherInstructions.enabled': teacherInstructions.settings &&
                toggle(teacherInstructions.label),
                'configure.studentInstructions.enabled': studentInstructions.settings &&
                toggle(studentInstructions.label),
              },
            }}
          />
        }
      >
        <div>
          <Typography className={classes.header} variant="subheading">
            Display
          </Typography>

          {width.settings &&
            <NumberTextField
              label={width.label}
              disabled={!this.state.setDimensions}
              value={parseInt(model.width)}
              min={100}
              max={500}
              onChange={this.changeWidth}
              showErrorWhenOutsideRange
              className={classes.field}
            />
          }
          {
            height.settings &&
              <NumberTextField
                label={height.label}
                disabled={!this.state.setDimensions}
                value={parseInt(model.height)}
                min={100}
                max={500}
                onChange={this.changeHeight}
                showErrorWhenOutsideRange
                className={classes.field}
              />
          }
          <br />
          <InputContainer label="Prompt" className={classes.promptContainer}>
            <EditableHtml markup={model.prompt} onChange={this.onPromptChange} />
          </InputContainer>

          <Typography className={classes.header} variant="subheading">
            Feedback
          </Typography>

          <FeedbackSelector
            label="When submitted, show"
            feedback={model.feedback || defaultFeedback}
            onChange={this.changeFeedback}
          />
        </div>
      </layout.ConfigLayout>
    );
  }
}
export default withStyles(theme => ({
  header: {
    paddingBottom: theme.spacing.unit
  },
  promptContainer: {
    paddingTop: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 2,
    width: '100%'
  },
  promptInput: {
    width: '100%',
    marginBottom: theme.spacing.unit
  },
  field: {
    width: '200px'
  }
}))(Main);
