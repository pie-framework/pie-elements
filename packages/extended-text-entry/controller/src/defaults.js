export default {
  prompt: '<div>This is the question prompt</div>',
  width: '500px',
};
