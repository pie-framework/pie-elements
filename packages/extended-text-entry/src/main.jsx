import React from 'react';
import {EditableHtml} from '@pie-lib/pie-toolbox/editable-html';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import debug from 'debug';
import debounce from 'lodash/debounce';
import { color, Feedback, Collapsible, PreviewPrompt } from '@pie-lib/pie-toolbox/render-ui';
import classnames from 'classnames';

const log = debug('@pie-ui:extended-text-entry');

const style = (theme) => ({
  main: {
    backgroundColor: color.background(),
    color: color.text(),
  },
  prompt: {
    width: '100%',
    color: color.text(),
    marginBottom: theme.spacing.unit * 2,
    fontSize: 'inherit',
  },
  teacherInstructions: {
    marginBottom: theme.spacing.unit * 2,
  },
  editor: {
    marginBottom: theme.spacing.unit * 2,
    borderRadius: '4px',
  },
  srOnly: {
    position: 'absolute',
    left: '-10000px',
    top: 'auto',
    width: '1px',
    height: '1px',
    overflow: 'hidden',
  },
});

export class Main extends React.Component {
  static propTypes = {
    onChange: PropTypes.func.isRequired,
    model: PropTypes.object,
    classes: PropTypes.object.isRequired,
    session: PropTypes.shape({
      value: PropTypes.string,
    }).isRequired,
  };

  changeSession = debounce(this.props.onChange, 1500);

  render() {
    const { model, classes, session } = this.props;

    const {
      dimensions,
      disabled,
      feedback,
      teacherInstructions,
      mathInput,
      spanishInput,
      specialInput,
      animationsDisabled,
      playersToolbarPosition,
      spellCheckEnabled,
    } = model;
    const { value } = session;
    const { width, height } = dimensions || {};
    const maxHeight = '40vh';
    const toolbarOpts = { position: playersToolbarPosition === 'top' ? 'top' : 'bottom' };

    log('[render] disabled? ', disabled);

    const teacherInstructionsDiv = (
      <PreviewPrompt defaultClassName="teacher-instructions" prompt={teacherInstructions} />
    );

    const languageCharactersProps = [];

    if (spanishInput) {
      languageCharactersProps.push({ language: 'spanish' });
    }

    if (specialInput) {
      languageCharactersProps.push({ language: 'special' });
    }

    return (
      <div
        className={classes.main}
        ref={(ref) => {
          this.containerRef = ref;
        }}
      >
        <h2 className={classes.srOnly}>Constructed Response Question</h2>

        {teacherInstructions && (
          <div className={classes.teacherInstructions}>
            {!animationsDisabled ? (
              <Collapsible
                labels={{ hidden: 'Show Teacher Instructions', visible: 'Hide Teacher Instructions' }}
                className={classes.collapsible}
              >
                {teacherInstructionsDiv}
              </Collapsible>
            ) : (
              teacherInstructionsDiv
            )}
          </div>
        )}

        {model.prompt && (
          <Typography component={'span'} className={classes.prompt}>
            <PreviewPrompt defaultClassName="prompt" prompt={model.prompt} />
          </Typography>
        )}

        <EditableHtml
          className={classnames(classes.editor, 'response-area-editor')}
          onChange={this.changeSession}
          markup={value || ''}
          width={width && width.toString()}
          minHeight={height && height.toString()}
          maxHeight={maxHeight}
          disabled={disabled}
          highlightShape={true}
          toolbarOpts={toolbarOpts}
          spellCheck={spellCheckEnabled}
          charactersLimit={50000}
          pluginProps={{
            math: {
              disabled: !mathInput,
              customKeys: this.props.model.customKeys,
              keypadMode: this.props.model.equationEditor,
              controlledKeypadMode: false,
            },
            video: {
              disabled: true,
            },
            audio: {
              disabled: true,
            },
          }}
          languageCharactersProps={languageCharactersProps}
        />

        {feedback && <Feedback correctness="correct" feedback={feedback} />}
      </div>
    );
  }
}

export default withStyles(style)(Main);
