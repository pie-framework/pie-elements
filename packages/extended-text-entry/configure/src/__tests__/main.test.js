import React from 'react';
import { Main } from '../main';
import { shallow } from 'enzyme';

import {
  layout,
  settings,
} from '@pie-lib/config-ui';

import defaults from '../defaults';

jest.mock('@pie-lib/config-ui', () => ({
  layout: {
    ConfigLayout: props => <div>{props.children}</div>
  },
  settings: {
    Panel: props => <div onChange={props.onChange} />,
    toggle: jest.fn(),
    radio: jest.fn()
  }
}));

describe('Render Main Component', () => {
  let wrapper, instance, onChange;
  let model = defaults.model;
  let configuration = defaults.configuration;

  beforeEach(() => {
    onChange = jest.fn();
    wrapper = shallow(
      <Main
        classes={{}}
        model={model}
        configuration={configuration}
        onModelChanged={onChange}
        handleBoxResize={() => {}}
      />
    );

    instance = wrapper.instance();
  });

  it('Match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('logic', () => {
    it('changeWidth calls onModelChanged', () => {
      instance.changeWidth({}, 10);
      expect(onChange).toBeCalledWith(expect.objectContaining({ width: 10 }));
    });

    it('changeHeight call onModelChanged', () => {
      instance.changeHeight({}, 10);
      expect(onChange).toBeCalledWith(expect.objectContaining({ height: 10 }));
    });

    it('changeTeacherInstructions calls onModelChanged', () => {
      instance.changeTeacherInstructions('Teacher Instructions');
      expect(onChange).toBeCalledWith(expect.objectContaining({ teacherInstructions: 'Teacher Instructions' }));
    });
  });
});
