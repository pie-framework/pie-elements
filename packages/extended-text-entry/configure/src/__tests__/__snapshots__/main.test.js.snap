// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Main Component Match Snapshot 1`] = `
<ConfigLayout
  settings={
    <Panel
      groups={
        Object {
          "Item Type": Object {
            "configure.equationEditor.enabled": false,
            "configure.multipleParts.enabled": false,
            "mathInput": undefined,
          },
          "Properties": Object {
            "configure.studentInstructions.enabled": false,
            "configure.teacherInstructions.enabled": false,
          },
        }
      }
      model={
        Object {
          "configure": Object {
            "equationEditor": Object {
              "enabled": false,
              "label": "Equation Editor",
              "settings": false,
            },
            "height": Object {
              "label": "Height (px)",
              "settings": true,
            },
            "mathInput": Object {
              "enabled": false,
              "label": "Student response can include math notation",
              "settings": true,
            },
            "multipleParts": Object {
              "enabled": false,
              "label": "Multiple Parts",
              "settings": false,
            },
            "studentInstructions": Object {
              "enabled": true,
              "label": "Student Instructions",
              "settings": false,
            },
            "teacherInstructions": Object {
              "enabled": false,
              "label": "Teacher Instructions",
              "settings": false,
            },
            "width": Object {
              "label": "Width (px)",
              "settings": true,
            },
          },
          "height": 100,
          "mathInput": false,
          "prompt": "This is the question prompt",
          "width": 500,
        }
      }
      onChangeModel={[Function]}
    />
  }
>
  <div>
    <WithStyles(Typography)
      variant="subheading"
    >
      Display
    </WithStyles(Typography)>
    <Component
      disabled={false}
      label="Width (px)"
      max={500}
      min={100}
      onChange={[Function]}
      showErrorWhenOutsideRange={true}
      value={500}
    />
    <Component
      disabled={false}
      label="Height (px)"
      max={500}
      min={100}
      onChange={[Function]}
      showErrorWhenOutsideRange={true}
      value={100}
    />
    <br />
    <Component
      label="Prompt"
    >
      <EditableHtml
        markup="This is the question prompt"
        onChange={[Function]}
      />
    </Component>
    <WithStyles(Typography)
      variant="subheading"
    >
      Feedback
    </WithStyles(Typography)>
    <Component
      feedback={
        Object {
          "default": "Your answer has been submitted",
          "type": "default",
        }
      }
      label="When submitted, show"
      onChange={[Function]}
    />
  </div>
</ConfigLayout>
`;
